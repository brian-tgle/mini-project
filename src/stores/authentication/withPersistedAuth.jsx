import React, { useLayoutEffect, useState } from 'react';
import { AuthenticationContainer, initState as initialStoreState, storeKey } from './authentication';
import databases from '../../cache';

const withPersistedAuth = (Component) => ({ ...props }) => {
  const [storePersisted, setStorePersisted] = useState(initialStoreState);

  useLayoutEffect(() => {
    (async () => {
      const data = await databases.getItem(storeKey).catch((error) => {
        // eslint-disable-next-line no-console
        console.error(error);
      });
      if (data) {
        setStorePersisted({
          ...data,
          initiated: true
        });
      } else {
        setStorePersisted({
          ...initialStoreState,
          initiated: true
        });
      }
    })();
  }, []);
  if (storePersisted && !storePersisted.initiated) return null;
  return (
    <AuthenticationContainer isGlobal initialState={storePersisted}>
      <Component {...props} />
    </AuthenticationContainer>
  );
};

export default withPersistedAuth;
