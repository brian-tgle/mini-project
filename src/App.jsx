import React, { Suspense } from 'react';
import {
  BrowserRouter, Redirect, Route, Switch
} from 'react-router-dom';
import Login from 'views/Login';
import AdminLayout from 'layouts/Admin';
import 'stores/middlewares/persistent';
import withPersistedAuth from 'stores/authentication/withPersistedAuth';
import Loading from 'components/Loading/Loading';

const App = () => (
  <BrowserRouter>
    <Suspense fallback={<Loading />}>
      <Switch>
        <Route path="/login" component={Login} />
        <Route path="/admin" render={(props) => <AdminLayout {...props} />} />
        <Redirect from="/" to="/admin/dashboard" />
      </Switch>
    </Suspense>
  </BrowserRouter>
);

export default withPersistedAuth(App);
