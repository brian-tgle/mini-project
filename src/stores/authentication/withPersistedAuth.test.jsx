import testSnapshots from 'ultis/test';
import withPersistedAuth from './withPersistedAuth';
import App from '../../App';

describe('withPersistedAuth', () => {
  const wrapper = withPersistedAuth(App);
  testSnapshots(wrapper, [
    {
      description: 'render snapshot',
      props: {}
    }
  ]);
});
