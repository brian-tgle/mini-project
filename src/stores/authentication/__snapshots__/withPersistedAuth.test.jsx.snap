// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPersistedAuth snapshots render snapshot 1`] = `null`;
